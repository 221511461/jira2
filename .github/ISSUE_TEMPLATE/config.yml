blank_issues_enabled: false
contact_links:
  - name: Support and Bug Requests
    url: https://support.atlassian.com/jira-software-cloud/
    about: Need some help? Please submit a support request or bug report via Jira Software Support portal
