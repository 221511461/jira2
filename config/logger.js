const bunyan = require('bunyan');
const bformat = require('bunyan-format');
// add levelInString to include DEBUG | ERROR | INFO | WARN
const formatOut = bformat({ outputMode: 'bunyan', levelInString: true });

const logger = bunyan.createLogger(
  {
    name: 'github-for-jira',
    stream: formatOut,
    level: 'info',
    serializers: {
      req: require('bunyan-express-serializer'),
    },
  },
);

// Suppress logging in tests
if (process.env.NODE_ENV === 'test') {
  logger.level(bunyan.FATAL + 1);
}

console.debug = logger.debug.bind(logger);
console.error = logger.error.bind(logger);
console.log = logger.info.bind(logger);
console.warn = logger.warn.bind(logger);

module.exports = logger;
