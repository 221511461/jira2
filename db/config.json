{
  "development": {
    "database": "jira-dev",
    "host": "127.0.0.1",
    "dialect": "postgres"
  },
  "test": {
    "database": "jira-test",
    "host": "127.0.0.1",
    "dialect": "postgres"
  },
  "production": {
    "dialect": "postgres",
    "use_env_variable": "PG_DATABASE_URL",
    "disable_sql_logging": true,
    "ssl": false,
    "dialectOptions": {
      "ssl": false
    },
    "pool": {
      "max": 15
    }
  }
}
