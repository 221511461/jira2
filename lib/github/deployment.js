const { Project } = require('../models');
const transformDeployment = require('../transforms/deployment');
const reduceProjectKeys = require('../jira/util/reduce-project-keys');

module.exports = async (context, jiraClient) => {
  if (context.payload.sender.login === 'github-actions[bot]') {
    const { data: jiraPayload } = await transformDeployment(context);

    if (!jiraPayload) {
      context.log({ noop: 'no_jira_payload_deployment' }, 'Halting further execution for deployment since jiraPayload is empty');
      return;
    }

    await jiraClient.deployment.submit(jiraPayload);

    const projects = [];
    reduceProjectKeys(jiraPayload, projects);

    for (const projectKey of projects) {
      await Project.upsert(projectKey, jiraClient.baseURL);
    }
  } else {
    // Solution for releasing MVP of GitHub Actions and avoid data duplication in Jira.
    context.log.warn('Deployment was not initiated by GitHub, skipping it.');
  }
};
