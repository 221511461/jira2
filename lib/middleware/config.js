const getConfig = require('probot-config')

module.exports = async (context) => {
  context.config = await getConfig(context, 'jira.yml')

  return context
}
