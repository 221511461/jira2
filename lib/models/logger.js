const bunyan = require('bunyan')
const bformat = require('bunyan-format')

module.exports = bunyan.createLogger({
  name: 'sequelize',
  level: 'debug',
  stream: bformat({ outputMode: 'short' })
})
