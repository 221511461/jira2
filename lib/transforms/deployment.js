const parseSmartCommit = require('./smart-commit');

function mapState({ state }) {
  // jira: unknown, pending, in_progress, cancelled, failed, rolled_back, successful
  // gh: error, failure, inactive, in_progress, queued, pending, success
  if (state === 'queued') {
    return 'pending';
  } else if (state === 'pending' || state === 'in_progress') {
    return 'in_progress';
  } else if (state === 'success') {
    return 'successful';
  } else if (state === 'error' || state === 'failure') {
    return 'failed';
  } else if (state === 'inactive') {
    return 'rolled_back';
  } else {
    return 'unknown';
  }
}

function mapEnvironment({ environment }, isProdEnvironment) {
  // jira: unmapped, development, testing, staging, production
  // gh: free-form string
  if (isProdEnvironment === true) {
    return 'production';
  }

  const isEnvironment = (envName) =>
    environment.localeCompare(envName, undefined, { sensitivity: 'base', ignorePunctuation: true }) === 0;

  if (isEnvironment('development') || isEnvironment('dev') || isEnvironment('trunk')) {
    return 'development';
  } else if (isEnvironment('testing') || isEnvironment('test') || isEnvironment('tests')
    || isEnvironment('tst')
    || isEnvironment('integration') || isEnvironment('integ') || isEnvironment('intg')
    || isEnvironment('int')
    || isEnvironment('acceptance') || isEnvironment('accept') || isEnvironment('acpt')
    || isEnvironment('qa') || isEnvironment('qc') || isEnvironment('control')
    || isEnvironment('quality')) {
    return 'testing';
  } else if (isEnvironment('staging') || isEnvironment('stage') || isEnvironment('stg')
    || isEnvironment('preprod') || isEnvironment('model') || isEnvironment('internal')) {
    return 'staging';
  } else if (isEnvironment('production') || isEnvironment('prod') || isEnvironment('live')) {
    return 'production';
  } else {
    return 'unmapped';
  }
}

module.exports = async (context) => {
  const { github, payload: { deployment_status, deployment } } = context;
  const { data: { commit: { message } } } = await github.repos.getCommit(context.repo({ ref: deployment.sha }));
  const { issueKeys } = parseSmartCommit(`${deployment.ref}\n${message}`);

  if (!issueKeys) {
    return { data: undefined };
  }

  return {
    data: {
      deployments: [{
        schemaVersion: '1.0',
        deploymentSequenceNumber: deployment.id,
        updateSequenceNumber: deployment_status.id,
        issueKeys,
        displayName: deployment.task,
        url: deployment_status.log_url || deployment_status.target_url,
        description: deployment.description || deployment_status.description || deployment.task,
        lastUpdated: deployment_status.updated_at,
        state: mapState(deployment_status),
        pipeline: {
          id: deployment.task,
          displayName: deployment.task,
          url: deployment_status.log_url || deployment_status.target_url,
        },
        environment: {
          id: deployment_status.environment,
          displayName: deployment_status.environment,
          type: mapEnvironment(deployment_status, deployment.production_environment),
        },
      }],
    },
  };
};
