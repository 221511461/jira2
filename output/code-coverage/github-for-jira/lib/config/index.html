
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for github-for-jira/lib/config</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> github-for-jira/lib/config</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">98.67% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>74/75</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">59.09% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>13/22</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>16/16</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">98.61% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>71/72</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="cidr-validator.js"><a href="cidr-validator.js.html">cidr-validator.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	</tr>

<tr>
	<td class="file high" data-value="enhance-octokit.js"><a href="enhance-octokit.js.html">enhance-octokit.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	</tr>

<tr>
	<td class="file high" data-value="github-api.js"><a href="github-api.js.html">github-api.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="14" class="abs high">14/14</td>
	<td data-value="71.43" class="pct medium">71.43%</td>
	<td data-value="7" class="abs medium">5/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="14" class="abs high">14/14</td>
	</tr>

<tr>
	<td class="file high" data-value="redis-info.js"><a href="redis-info.js.html">redis-info.js</a></td>
	<td data-value="87.5" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 87%"></div><div class="cover-empty" style="width: 13%"></div></div>
	</td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="8" class="abs high">7/8</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="10" class="abs medium">6/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="7" class="abs high">6/7</td>
	</tr>

<tr>
	<td class="file high" data-value="statsd.js"><a href="statsd.js.html">statsd.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="20" class="abs high">20/20</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="19" class="abs high">19/19</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Thu Apr 01 2021 15:06:52 GMT+1100 (Australian Eastern Daylight Time)
            </div>
        </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    