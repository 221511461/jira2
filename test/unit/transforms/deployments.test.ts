/* eslint-disable @typescript-eslint/no-var-requires */
// eslint-disable-next-line import/no-duplicates
import { mapEnvironment } from "../../../src/transforms/deployment";
import transformDeployment from "../../../src/transforms/deployment";
import {getLogger} from "../../../src/config/logger";
import {GitHubAPI} from "probot";
import { when } from "jest-when";
import { booleanFlag, BooleanFlags } from "../../../src/config/feature-flags";

jest.mock("../../../src/config/feature-flags");

describe("deployment environment mapping", () => {
	test("classifies known environments correctly", () => {
		// Development
		expect(mapEnvironment("development")).toBe("development");
		expect(mapEnvironment("dev")).toBe("development");
		expect(mapEnvironment("trunk")).toBe("development");

		// Testing
		expect(mapEnvironment("testing")).toBe("testing");
		expect(mapEnvironment("test")).toBe("testing");
		expect(mapEnvironment("tests")).toBe("testing");
		expect(mapEnvironment("tst")).toBe("testing");
		expect(mapEnvironment("integration")).toBe("testing");
		expect(mapEnvironment("integ")).toBe("testing");
		expect(mapEnvironment("intg")).toBe("testing");
		expect(mapEnvironment("int")).toBe("testing");
		expect(mapEnvironment("acceptance")).toBe("testing");
		expect(mapEnvironment("accept")).toBe("testing");
		expect(mapEnvironment("acpt")).toBe("testing");
		expect(mapEnvironment("qa")).toBe("testing");
		expect(mapEnvironment("qc")).toBe("testing");
		expect(mapEnvironment("control")).toBe("testing");
		expect(mapEnvironment("quality")).toBe("testing");

		// Staging
		expect(mapEnvironment("staging")).toBe("staging");
		expect(mapEnvironment("stage")).toBe("staging");
		expect(mapEnvironment("stg")).toBe("staging");
		expect(mapEnvironment("preprod")).toBe("staging");
		expect(mapEnvironment("model")).toBe("staging");
		expect(mapEnvironment("internal")).toBe("staging");

		// Production
		expect(mapEnvironment("production")).toBe("production");
		expect(mapEnvironment("prod")).toBe("production");
		expect(mapEnvironment("prd")).toBe("production");
		expect(mapEnvironment("live")).toBe("production");
	});

	test("classifies known environments with prefixes and/or postfixes correctly", () => {
		expect(mapEnvironment("prod-east")).toBe("production");
		expect(mapEnvironment("prod_east")).toBe("production");
		expect(mapEnvironment("east-staging")).toBe("staging");
		expect(mapEnvironment("qa:1")).toBe("testing");
		expect(mapEnvironment("mary-dev:1")).toBe("development");
		expect(mapEnvironment("スパイク・スピーゲル-dev:1")).toBe("development");
		expect(mapEnvironment("trunk alpha")).toBe("development");
		expect(mapEnvironment("production(us-east)")).toBe("production");
		expect(mapEnvironment("prd (eu-central)")).toBe("production");
	});

	test("ignores case", () => {
		expect(mapEnvironment("Staging")).toBe("staging");
		expect(mapEnvironment("PROD-east")).toBe("production");
	});

	test("ignores diacritics", () => {
		expect(mapEnvironment("stàging")).toBe("staging");
	});

	test("classifies unknown environment names as 'unmapped'", () => {
		expect(mapEnvironment("banana-east")).toBe("unmapped");
		expect(mapEnvironment("internet")).toBe("unmapped");
		expect(mapEnvironment("製造")).toBe("unmapped");
	});
});

describe("transform GitHub webhook payload to Jira payload", () => {

	const deployment_status = require("../../fixtures/deployment_status-basic.json");
	const owner = deployment_status.payload.repository.owner.login;
	const repo = deployment_status.payload.repository.name;

	it("supports branch and merge workflows - FF TRUE", async () => {
		// Mocking all GitHub API Calls
		// Get commit
		githubNock.get(`/repos/${owner}/${repo}/commits/${deployment_status.payload.deployment.sha}`)
			.reply(200, {
				...owner,
				commit: {
					message: "testing"
				}
			});

		// List deployments
		githubNock.get(`/repos/${owner}/${repo}/deployments?environment=Production&per_page=10`)
			.reply(200,
				[
					{
						id: 1,
						environment: "Production",
						sha: "6e87a40179eb7ecf5094b9c8d690db727472d5bc"
					}
				]
			);

		// List deployments statuses
		githubNock.get(`/repos/${owner}/${repo}/deployments/1/statuses?per_page=100`)
			.reply(200, [
				{
					id: 1,
					state: "pending"
				},
				{
					id: 2,
					state: "success"
				}
			]);

		// Compare commits
		githubNock.get(`/repos/${owner}/${repo}/compare/6e87a40179eb7ecf5094b9c8d690db727472d5bc...${deployment_status.payload.deployment.sha}`)
			.reply(200,{
				commits: [
					{
						commit: {
							message: "ABC-1"
						}
					},
					{
						commit: {
							message: "ABC-2"
						}
					}
				]}
			);


		when(booleanFlag).calledWith(
			BooleanFlags.SUPPORT_BRANCH_AND_MERGE_WORKFLOWS_FOR_DEPLOYMENTS,
			expect.anything(),
			expect.anything()
		).mockResolvedValue(true);

		const jiraPayload = await transformDeployment(GitHubAPI(), deployment_status.payload, "testing.atlassian.net", getLogger("deploymentLogger"))

		expect(jiraPayload).toMatchObject({
			deployments: [{
				schemaVersion: "1.0",
				deploymentSequenceNumber: 1234,
				updateSequenceNumber: 123456,
				issueKeys: ["ABC-1", "ABC-2"],
				displayName: "deploy",
				url: "test-repo-url/commit/885bee1-commit-id-1c458/checks",
				description: "deploy",
				lastUpdated: new Date("2021-06-28T12:15:18.000Z"),
				state: "successful",
				pipeline: {
					id: "deploy",
					displayName: "deploy",
					url: "test-repo-url/commit/885bee1-commit-id-1c458/checks",
				},
				environment: {
					id: "Production",
					displayName: "Production",
					type: "production",
				},
			}]
		})
	})

	// TODO add test if FF is false
});
