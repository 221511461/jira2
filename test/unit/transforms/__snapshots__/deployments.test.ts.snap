// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deployment environment mapping should transform commonly used environment names to the expected environment types 1`] = `
Object {
  "development": Array [
    "development",
    "DEVELOPMENT",
    "development-east",
    "development-eu-west",
    "east-development",
    "jane-development-local",
    "development:east",
    "development_east",
    "development/east",
    "development east",
    "dev",
    "DEV",
    "dev-east",
    "dev-eu-west",
    "east-dev",
    "jane-dev-local",
    "dev:east",
    "dev_east",
    "dev/east",
    "dev east",
    "trunk",
    "TRUNK",
    "trunk-east",
    "trunk-eu-west",
    "east-trunk",
    "jane-trunk-local",
    "trunk:east",
    "trunk_east",
    "trunk/east",
    "trunk east",
    "dev",
    "DEV",
    "dev-east",
    "dev-eu-west",
    "east-dev",
    "jane-dev-local",
    "dev:east",
    "dev_east",
    "dev/east",
    "dev east",
  ],
  "production": Array [
    "production",
    "PRODUCTION",
    "production-east",
    "production-eu-west",
    "east-production",
    "jane-production-local",
    "production:east",
    "production_east",
    "production/east",
    "production east",
    "prod",
    "PROD",
    "prod-east",
    "prod-eu-west",
    "east-prod",
    "jane-prod-local",
    "prod:east",
    "prod_east",
    "prod/east",
    "prod east",
    "live",
    "LIVE",
    "live-east",
    "live-eu-west",
    "east-live",
    "jane-live-local",
    "live:east",
    "live_east",
    "live/east",
    "live east",
  ],
  "staging": Array [
    "staging",
    "STAGING",
    "staging-east",
    "staging-eu-west",
    "east-staging",
    "jane-staging-local",
    "staging:east",
    "staging_east",
    "staging/east",
    "staging east",
    "stage",
    "STAGE",
    "stage-east",
    "stage-eu-west",
    "east-stage",
    "jane-stage-local",
    "stage:east",
    "stage_east",
    "stage/east",
    "stage east",
    "stg",
    "STG",
    "stg-east",
    "stg-eu-west",
    "east-stg",
    "jane-stg-local",
    "stg:east",
    "stg_east",
    "stg/east",
    "stg east",
    "preprod",
    "PREPROD",
    "preprod-east",
    "preprod-eu-west",
    "east-preprod",
    "jane-preprod-local",
    "preprod:east",
    "preprod_east",
    "preprod/east",
    "preprod east",
    "model",
    "MODEL",
    "model-east",
    "model-eu-west",
    "east-model",
    "jane-model-local",
    "model:east",
    "model_east",
    "model/east",
    "model east",
    "internal",
    "INTERNAL",
    "internal-east",
    "internal-eu-west",
    "east-internal",
    "jane-internal-local",
    "internal:east",
    "internal_east",
    "internal/east",
    "internal east",
  ],
  "testing": Array [
    "testing",
    "TESTING",
    "testing-east",
    "testing-eu-west",
    "east-testing",
    "jane-testing-local",
    "testing:east",
    "testing_east",
    "testing/east",
    "testing east",
    "test",
    "TEST",
    "test-east",
    "test-eu-west",
    "east-test",
    "jane-test-local",
    "test:east",
    "test_east",
    "test/east",
    "test east",
    "tests",
    "TESTS",
    "tests-east",
    "tests-eu-west",
    "east-tests",
    "jane-tests-local",
    "tests:east",
    "tests_east",
    "tests/east",
    "tests east",
    "tst",
    "TST",
    "tst-east",
    "tst-eu-west",
    "east-tst",
    "jane-tst-local",
    "tst:east",
    "tst_east",
    "tst/east",
    "tst east",
    "integration",
    "INTEGRATION",
    "integration-east",
    "integration-eu-west",
    "east-integration",
    "jane-integration-local",
    "integration:east",
    "integration_east",
    "integration/east",
    "integration east",
    "integ",
    "INTEG",
    "integ-east",
    "integ-eu-west",
    "east-integ",
    "jane-integ-local",
    "integ:east",
    "integ_east",
    "integ/east",
    "integ east",
    "intg",
    "INTG",
    "intg-east",
    "intg-eu-west",
    "east-intg",
    "jane-intg-local",
    "intg:east",
    "intg_east",
    "intg/east",
    "intg east",
    "int",
    "INT",
    "int-east",
    "int-eu-west",
    "east-int",
    "jane-int-local",
    "int:east",
    "int_east",
    "int/east",
    "int east",
    "acceptance",
    "ACCEPTANCE",
    "acceptance-east",
    "acceptance-eu-west",
    "east-acceptance",
    "jane-acceptance-local",
    "acceptance:east",
    "acceptance_east",
    "acceptance/east",
    "acceptance east",
    "accept",
    "ACCEPT",
    "accept-east",
    "accept-eu-west",
    "east-accept",
    "jane-accept-local",
    "accept:east",
    "accept_east",
    "accept/east",
    "accept east",
    "acpt",
    "ACPT",
    "acpt-east",
    "acpt-eu-west",
    "east-acpt",
    "jane-acpt-local",
    "acpt:east",
    "acpt_east",
    "acpt/east",
    "acpt east",
    "qa",
    "QA",
    "qa-east",
    "qa-eu-west",
    "east-qa",
    "jane-qa-local",
    "qa:east",
    "qa_east",
    "qa/east",
    "qa east",
    "qc",
    "QC",
    "qc-east",
    "qc-eu-west",
    "east-qc",
    "jane-qc-local",
    "qc:east",
    "qc_east",
    "qc/east",
    "qc east",
    "control",
    "CONTROL",
    "control-east",
    "control-eu-west",
    "east-control",
    "jane-control-local",
    "control:east",
    "control_east",
    "control/east",
    "control east",
    "quality",
    "QUALITY",
    "quality-east",
    "quality-eu-west",
    "east-quality",
    "jane-quality-local",
    "quality:east",
    "quality_east",
    "quality/east",
    "quality east",
  ],
  "unmapped": Array [
    "some-random-name",
    "SOME-RANDOM-NAME",
    "some-random-name-east",
    "some-random-name-eu-west",
    "east-some-random-name",
    "jane-some-random-name-local",
    "some-random-name:east",
    "some-random-name_east",
    "some-random-name/east",
    "some-random-name east",
    "internet",
    "INTERNET",
    "internet-east",
    "internet-eu-west",
    "east-internet",
    "jane-internet-local",
    "internet:east",
    "internet_east",
    "internet/east",
    "internet east",
  ],
}
`;
